src/main/java/com/liferay/faces/util/jsp/internal/PageAdapter.java

```diff
@@ -16,23 +16,28 @@
 package com.liferay.faces.util.jsp.internal;
 
 import java.io.IOException;
+import java.util.Collections;
+import java.util.Enumeration;
 
-import javax.servlet.Servlet;
 import javax.servlet.ServletConfig;
+import javax.servlet.ServletContext;
 import javax.servlet.ServletException;
 import javax.servlet.ServletRequest;
 import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServlet;
 
 
 /**
  * @author  Neil Griffin
  */
-public class PageAdapter implements Servlet {
+public class PageAdapter extends HttpServlet {
 
 	private ServletConfig servletConfig;
+	private ServletContext servletContext;
 
-	public PageAdapter(ServletConfig servletConfig) {
-		this.servletConfig = servletConfig;
+	public PageAdapter(ServletContext servletContext) {
+		this.servletConfig = new PageAdapterServletConfig();
+		this.servletContext = servletContext;
```

Comment:
I've adapted this class for the needs in this issue. I've been searching and I couldn't find the place where it's used.